standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -1266,7 +1309,7 @@ public Table getTable(String catName, String dbName, String tblName, String vali
     catName = normalizeIdentifier(catName);
     dbName = StringUtils.normalizeIdentifier(dbName);
     tblName = StringUtils.normalizeIdentifier(tblName);
-    if (!shouldCacheTable(catName, dbName, tblName) || (canUseEvents && rawStore.isActiveTransaction())) {
+    if (shouldGetTableFromRawStore(catName, dbName, tblName, validWriteIds)) {
```

Comment:
We should use "shouldGetTableFromRawStore" even for get partitions, column stats.

Reply:
I will do that in new PR. Where I will put tableId and writeId check on all the methods in cacheStore. This PR is mostly around adding validWriteIdList to TableWrapper Object in cacheStore.